Ginger/GingerCoreNET/Run/RunSetActions/RunSetActionPublishToQCOperations.cs

```diff
@@ -82,5 +119,57 @@ public void Execute(IReportInfo RI)
             }
         }
 
+        private BusinessFlow ConvertRunSetToBF(IReportInfo reportInfo)
+        {
+            RunsetExecutor runSetExec = WorkSpace.Instance.RunsetExecutor;
+            try
+            {
+                if (reportInfo == null)
+                {
+                    return null;
+                }
+
+                //Create Business Flow
+                BusinessFlow virtualBF = new BusinessFlow();
+                virtualBF.Name = runSetExec.RunSetConfig.Name;
+                virtualBF.Description = runSetExec.RunSetConfig.Description;
+                virtualBF.Status = BusinessFlow.eBusinessFlowStatus.Development;
+                virtualBF.RunStatus = runSetExec.RunSetConfig.RunSetExecutionStatus;
+                virtualBF.Activities = new ObservableList<Activity>();
+                foreach (GingerRunner runSetrunner in runSetExec.Runners)
+                {
+                    // if executor is null when run if from file
+                    if(runSetrunner.Executor is null)
+                    {
+                        runSetrunner.Executor = new GingerExecutionEngine(runSetrunner);
+                    }
+                    foreach (BusinessFlow runSetBF in runSetrunner.Executor.BusinessFlows)
+                    {
+                        ActivitiesGroup virtualAG = new ActivitiesGroup();
+                        virtualAG.Name = runSetBF.Name;
+                        virtualAG.Description = runSetBF.Description;
+                        if (Enum.IsDefined(typeof(eActivitiesGroupRunStatus), runSetBF.RunStatus.ToString()))
+                        {
+                            virtualAG.RunStatus = (eActivitiesGroupRunStatus)Enum.Parse(typeof(eActivitiesGroupRunStatus), runSetBF.RunStatus.ToString());
+                        }
+                        else
+                        {
+                            virtualAG.RunStatus = eActivitiesGroupRunStatus.NA;
+                        }
+                        virtualBF.AddActivitiesGroup(virtualAG);
+                        foreach (Activity runSetAct in runSetBF.Activities)
+                        {
+                            virtualBF.AddActivity(runSetAct, virtualAG);
+                        }
+                    }
+                }
+                return virtualBF;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to Export ALM test set and convert it into " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow), ex);
```

Comment:
set error to "Failed to convert Run set to BF for ALM Export"

Reply:
done